dash/dash.py

```diff
@@ -427,8 +420,23 @@ def __init__(
 
         self.logger.setLevel(logging.INFO)
 
-    def init_app(self, app=None):
+    def init_app(self, app=None, **kwargs):
         """Initialize the parts of Dash that require a flask app."""
+
+        self.config.update(kwargs)
+        self.config.set_read_only(
+            [
+                "url_base_pathname",
+                "routes_pathname_prefix",
+                "requests_pathname_prefix",
+            ],
+            "Read-only: can only be set in the Dash constructor or during init_app()",
+        )
+
+        self.config.finalize(
+            "Invalid config key. Some settings are only available "
+            "via the Dash constructor"
+        )
```

Comment:
Can we leave `finalize` where it was? If it's here, users could add random items to config before `init_app` is called, but I believe all the valid keys have already been initialized in the `AttributeDict` constructor.

Alternatively we could get rid of `finalize` altogether and make a custom constructor for `AttributeDict` that bakes this in.

Reply:
I agree with your first point - I previously misunderstood finalize as "no further changes can be made to the `AttributeDict`". Having re-read the `__setitem__()` implementation this should be moved back where it was.

Reply:
OK great! So let's move it back, then add a changelog entry for this and it'll be ready to go - thanks!

Reply:
Updated `dash.py`, added a changelog entry and merged with the latest `dev` branch as of c50cbf4